e2e/docker/docker.go

```diff
@@ -118,6 +120,118 @@ func (c ctx) testDockerPulls(t *testing.T) {
 	}
 }
 
+// Testing DOCKER_ host support (only if docker available)
+func (c ctx) testDockerHost(t *testing.T) {
+	require.Command(t, "docker")
+
+	// Write a temporary "empty" Dockerfile (from scratch)
+	tmpPath, err := fs.MakeTmpDir(c.env.TestDir, "docker-", 0o755)
+	err = errors.Wrapf(err, "creating temporary directory in %q for docker host test", c.env.TestDir)
+	if err != nil {
+		t.Fatalf("failed to create temporary directory: %+v", err)
+	}
+	defer os.RemoveAll(tmpPath)
+
+	// Here is the Dockerfile, and a temporary image path
+	dockerfile := filepath.Join(tmpPath, "Dockerfile")
+	tmpImage := filepath.Join(tmpPath, "scratch-tmp.sif")
+
+	// Write to file
+	dockerfileContent := []byte("FROM scratch\n")
+	err = os.WriteFile(dockerfile, dockerfileContent, 0o644)
+	if err != nil {
+		t.Fatalf("failed to create temporary Dockerfile: %+v", err)
+	}
+
+	dockerURI := "docker-test-image:host"
+	pullURI := "docker-daemon://" + dockerURI
+
+	// Invoke docker build to build an empty scratch image in the docker daemon.
+	// Use os/exec because easier to generate a command with a working directory
+	cmd := exec.Command("docker", "build", dockerURI, "-noappend", "-all-root")
+	cmd.Dir = tmpPath
+	_, err = cmd.Output()
+	if err != nil {
+		t.Fatalf("Unexpected error while running command.\n%s", err)
+	}
+
+	tests := []struct {
+		name       string
+		envarName  string
+		envarValue string
+		exit       int
+	}{
+		// Unset docker host should use default and succeed
+		{
+			name:       "singularityDockerHostEmpty",
+			envarName:  "SINGULARITY_DOCKER_HOST",
+			envarValue: "",
+			exit:       0,
+		},
+		{
+			name:       "dockerHostEmpty",
+			envarName:  "DOCKER_HOST",
+			envarValue: "",
+			exit:       0,
+		},
+
+		// bad Docker host should fail
+		{
+			name:       "singularityDockerHostInvalid",
+			envarName:  "SINGULARITY_DOCKER_HOST",
+			envarValue: "tcp://192.168.59.103:oops",
+			exit:       255,
+		},
+		{
+			name:       "dockerHostInvalid",
+			envarName:  "DOCKER_HOST",
+			envarValue: "tcp://192.168.59.103:oops",
+			exit:       255,
+		},
+
+		// Set to default should succeed
+		// The default host varies based on OS, so we use dockerclient default
+		{
+			name:       "singularityDockerHostValid",
+			envarName:  "SINGULARITY_DOCKER_HOST",
+			envarValue: dockerclient.DefaultDockerHost,
+			exit:       255,
+		},
+		{
+			name:       "dockerHostValid",
+			envarName:  "DOCKER_HOST",
+			envarValue: dockerclient.DefaultDockerHost,
+			exit:       255,
+		},
+	}
+
+	for _, tt := range tests {
+
+		// Export to environment
+		if tt.envarValue != "" {
+			err := os.Setenv(tt.envarName, tt.envarValue)
```

Comment:
Don't export to the global environment, as it will stick around and could affect other things if not cleaned up carefully.

The `RunSingularity` function can take environment via an `e2e.WithEnv` call, that will be limited to our single run of singularity. See e.g. here:

https://github.com/sylabs/singularity/blob/f1477058a25cddedb0ca5b2377b6f9ef827865a7/e2e/run/run.go#L98